File: Check Armstrong Number/check_armstrong_number.c
#include <stdio.h>
#include <math.h>
int main() {
    int n, temp, digit, digits = 0, total = 0;
    printf("Enter a number: ");
    scanf("%d", &n);
    temp = n;
    while (temp != 0) {
        temp = temp / 10;
        digits++;
    }
    temp = n;
    while (temp != 0) {
        digit = temp % 10;
        total = total + (int) pow(digit, digits);
        temp = temp / 10;
    }
    if (total == n) {
        printf("%d is an Armstrong number\n", n);
    } else {
        printf("%d is not an Armstrong number\n", n);
    }
    return 0;
}

File: Check Armstrong Number/check_armstrong_number.py
n = int(input("Enter a number: "))
temp = n
digits = len(str(n))
total = 0
while temp > 0:
    digit = temp % 10
    total += digit ** digits
    temp //= 10
if total == n:
    print(n, "is an Armstrong number")
else:
    print(n, "is not an Armstrong number")
